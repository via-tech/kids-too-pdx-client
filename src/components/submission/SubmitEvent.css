.Form {
  display: block;
  margin: .5em;
  legend {
    text-align: center;
  }
  label, input {
    display: block;
    margin-top: 1em;
  }
  input[type="checkbox"], input[type="radio"] {
    display: inline;
  }
  textarea {
    width: 100%;
  }
  fieldset {
    margin-bottom: .5em;
  }
  button {
    display: block;
    margin: auto;
    padding: 1em;
    border-radius: 5px;
    background-color: white;
  }
  .required {
    color: red;
  }
}
.radio {
  display: flex;
  input {
    width: 1em;
    margin-left: .75em;
  }
}
@media only screen and (min-width: 1000px) {
  .Form {
    width: 75vw;
    margin: auto;
    fieldset {
      margin: auto;
      margin-top: 1em;
      margin-bottom: 1em;
    }
  }
}