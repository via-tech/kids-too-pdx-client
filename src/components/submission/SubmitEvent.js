import React from 'react';
import PropTypes from 'prop-types';
import styles from './SubmitEvent.css';

function SubmitEvent({ orgEvent, token, handleSubmit, handleChange, history }) {
  if(!token) {
    history.push('/login');
  }
  return (
    <>
      <form className={styles.Form} onSubmit={handleSubmit.bind(null, orgEvent)}>
        <fieldset>
          <legend>Event Contact Info</legend>

          <label>Name<span className={styles.required}>*</span>:
            <input type="text" name="contactName" onChange={handleChange} required/>
          </label>

          <label name="email">Email<span className={styles.required}>*</span>:
            <input type="email" name="email" onChange={handleChange} required/>
          </label>

          <label>Phone #<span className={styles.required}>*</span>:
            <input type="tel" name="phone" onChange={handleChange} required/>
          </label>
        </fieldset>
        <fieldset>
          <legend>Event Details</legend>

          <label>Event Name
            <input type="text" name="name" onChange={handleChange} />
          </label>

          <label>Registration Link
            <input type="text" name="website" onChange={handleChange}></input>
          </label>

          <label>Date<span className={styles.required}>*</span>:
            <input type="date" onChange={handleChange} name="date" required/>
          </label>

          <label>Time<span className={styles.required}>*</span>:
            <input type="string" onChange={handleChange} name="time" required />
          </label>

          <label>Location<span className={styles.required}>*</span>:
            <input type="text" placeholder="Venue" name="venue" onChange={handleChange} required />
            <input type="text" placeholder="Address" name="address" onChange={handleChange} required />
            <input type="text" placeholder="City" name="city" onChange={handleChange} />
            <input type="state" placeholder="State" name="state" onChange={handleChange} />
            <input type="zipcode" placeholder="Zipcode" name="zipcode" onChange={handleChange} />
          </label>

          <label>Price<span className={styles.required}>*</span>:
            <input type="number" placeholder="$" onChange={handleChange} name="price" min="0" required/>
          </label>

          <label className={styles.radio}>Reduced Rate Available:
            <input type="checkbox" name="reducedRate" value="1" onChange={handleChange}/>
          </label>

          <p>Age<span className={styles.required}>*</span></p>

          <input type="radio" onChange={handleChange} name="age" value="0" required/>All ages<br/>

          <input type="radio" onChange={handleChange} name="age" value="1"/>Enter an age range

          <label>
            <input type="number" placeholder="Min Age" onChange={handleChange} name="ageMin" min="0" max="100"/>
            
            <input type="number" placeholder="Max Age" onChange={handleChange} name="ageMax" min="0" max="100"/>
          </label>

          <label>Category<span className={styles.required}>*</span>
            <select onChange={handleChange} name="category" required>
              <option value="DEFAULT" hidden>Category</option>
              <option value="Sports">Sports</option>
              <option value="Art">Art</option>
              <option value="Music">Music</option>
              <option value="Performance">Performance</option>
              <option value="Educational">Educational</option>
              <option value="Volunteer">Volunteer</option>
            </select>
          </label>

          <label>Description<span className={styles.required}>*</span>
            <textarea onChange={handleChange} name="description" required/>
          </label>

          <label className={styles.radio}>By checking the box, you agree that your organization accepts all liability for this event.<span className={styles.required}>*</span>
            <input type="checkbox" value="1" name="liability" onChange={handleChange} required/>
          </label>

        </fieldset>
        <button type="submit">CREATE EVENT</button>
      </form>
    </>
  );
}

SubmitEvent.propTypes = {
  orgEvent: PropTypes.object.isRequired,
  token: PropTypes.string,
  handleChange: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  history: PropTypes.object
};

export default SubmitEvent;
