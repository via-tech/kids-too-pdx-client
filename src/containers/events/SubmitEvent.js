import { connect } from 'react-redux';
import React from 'react';
import SubmitEventForm from '../../components/submission/SubmitEventForm';
import { getEvent } from '../../selectors/submitEvent';
import { updateEvent, createEvent } from '../../actions/submitEvent';
import { getToken, getUser } from '../../selectors/session';

function SubmitEvent(props) {
  return <SubmitEventForm {...props} />;
}

const modelEvent = orgEvent => {
  const { contactName, email, phone, venue, address, city, state, zipcode, ageMin, ageMax } = orgEvent;

  return {
    ...orgEvent,
    contact: { contactName, email, phone },
    location: { venue, address, city, state, zipcode },
    ageMin: ageMin || 0,
    ageMax: ageMax || 100
  };
};

const mapStateToProps = state => ({
  orgEvent: getEvent(state),
  token: getToken(state),
  role: getUser(state).role
});

const mapDispatchToProps = (dispatch, props) => ({
  handleChange({ target }) {
    const { name, value } = target;
    dispatch(updateEvent({ [name]: value }));
  },

  handleSubmit(orgEvent, event) {
    const action = createEvent(modelEvent(orgEvent));
    event.preventDefault();
    dispatch(action);
    action.payload
      .then(({ _id }) => props.history.push(`/events/${_id}`));
  }
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(SubmitEvent);
