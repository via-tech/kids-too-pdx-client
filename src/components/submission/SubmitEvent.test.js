import React from 'react';
import renderer from 'react-test-renderer';
import SubmitEvent from './SubmitEvent';

describe('SubmitEvent', () => {
  const orgEvent = {
    _id: 1,
    image: 'image.com',
    time: '2pm',
    name: 'House Event',
    date: new Date(2019, 6, 1, 10, 30).toString(),
    contact: {
      contactName: 'Joe',
      email: 'joe@email.com',
      phone: '555-123-4567'
    },
    location: {
      venue: 'My House',
      address: '123 Main St.',
      city: 'Portland',
      state: 'OR',
      zipcode: 97223
    },
    price: 250,
    reducedRate: false,
    ageMin: 9,
    ageMax: 12,
    category: 'Art',
    description: 'The bestest event',
    liability: true
  };
  it('snapshots SubmitEvent logged in and active', () => {
    const props = {
      orgEvent,
      token: '1234',
      handleChange: jest.fn(),
      handleSubmit: jest.fn(),
      role: 'org'
    };

    const tree = renderer.create(
      <SubmitEvent {...props} />
    );

    expect(tree).toMatchSnapshot();
  });

  it('snapshots SubmitEvent logged out', () => {
    const props = {
      orgEvent,
      token: '',
      handleChange: jest.fn(),
      handleSubmit: jest.fn(),
      role: 'org',
      history: {
        push(route) {
          return route;
        }
      }
    };

    const tree = renderer.create(
      <SubmitEvent {...props} />
    );

    expect(tree).toMatchSnapshot();
  });

  it('snapshots SubmitEvent logged in and inactive org', () => {
    const props = {
      orgEvent,
      token: '1234',
      handleChange: jest.fn(),
      handleSubmit: jest.fn(),
      role: 'inactive',
      history: {
        push(route) {
          return route;
        }
      }
    };

    const tree = renderer.create(
      <SubmitEvent {...props} />
    );

    expect(tree).toMatchSnapshot();
  });
});
