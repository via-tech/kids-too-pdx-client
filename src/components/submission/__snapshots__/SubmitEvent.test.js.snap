// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmitEvent snapshots SubmitEvent logged in 1`] = `
<form
  className="Form"
  onSubmit={[Function]}
>
  <fieldset>
    <legend>
      Event Contact Info
    </legend>
    <label>
      Name:
      <input
        name="contactName"
        onChange={[MockFunction]}
        required={true}
        type="text"
      />
    </label>
    <label
      name="email"
    >
      Email:
      <input
        name="email"
        onChange={[MockFunction]}
        required={true}
        type="email"
      />
    </label>
    <label>
      Phone #:
      <input
        name="phone"
        onChange={[MockFunction]}
        required={true}
        type="tel"
      />
    </label>
  </fieldset>
  <fieldset>
    <legend>
      Event Details
    </legend>
    <label>
      Event Name:
      <input
        name="name"
        onChange={[MockFunction]}
        type="text"
      />
    </label>
    <label>
      Registration Link:
      <input
        name="website"
        onChange={[MockFunction]}
        type="text"
      />
    </label>
    <label>
      Date:
      <input
        name="date"
        onChange={[MockFunction]}
        required={true}
        type="date"
      />
    </label>
    <label>
      Time:
      <input
        name="time"
        onChange={[MockFunction]}
        required={true}
        type="string"
      />
    </label>
    <label>
      Location:
      <input
        name="venue"
        onChange={[MockFunction]}
        placeholder="Venue"
        required={true}
        type="text"
      />
      <input
        name="address"
        onChange={[MockFunction]}
        placeholder="Address"
        required={true}
        type="text"
      />
      <input
        name="city"
        onChange={[MockFunction]}
        placeholder="City"
        type="text"
      />
      <input
        name="state"
        onChange={[MockFunction]}
        placeholder="State"
        type="state"
      />
      <input
        name="zipcode"
        onChange={[MockFunction]}
        placeholder="Zipcode"
        type="zipcode"
      />
    </label>
    <label>
      Price:
      <input
        min="0"
        name="price"
        onChange={[MockFunction]}
        placeholder="$"
        required={true}
        type="number"
      />
    </label>
    <label
      className="radio"
    >
      Reduced Rate Available:
      <input
        name="reducedRate"
        onChange={[MockFunction]}
        type="checkbox"
        value=""
      />
    </label>
    <label>
      Age Range:
      <input
        min="0"
        name="minAge"
        onChange={[MockFunction]}
        placeholder="Min Age"
        required={true}
        type="number"
      />
      <input
        min="0"
        name="maxAge"
        onChange={[MockFunction]}
        placeholder="Max Age"
        required={true}
        type="number"
      />
    </label>
    <label>
      Category:
      <select
        name="category"
        onChange={[MockFunction]}
        required={true}
      >
        <option
          hidden={true}
          value="DEFAULT"
        >
          Category
        </option>
        <option
          value="Sports"
        >
          Sports
        </option>
        <option
          value="Art"
        >
          Art
        </option>
        <option
          value="Music"
        >
          Music
        </option>
        <option
          value="Performance"
        >
          Performance
        </option>
        <option
          value="Educational"
        >
          Educational
        </option>
        <option
          value="Volunteer"
        >
          Volunteer
        </option>
      </select>
    </label>
    <label>
      Description
      <textarea
        name="description"
        onChange={[MockFunction]}
        required={true}
      />
    </label>
    <label
      className="radio"
    >
      By checking the box, you agree that your organization accepts all liability for this event:
      <input
        name="liability"
        onChange={[MockFunction]}
        required={true}
        type="checkbox"
        value=""
      />
    </label>
  </fieldset>
  <button
    type="submit"
  >
    CREATE EVENT
  </button>
</form>
`;

exports[`SubmitEvent snapshots SubmitEvent logged out 1`] = `
<h1>
  Please 
  <a
    href="/login"
    onClick={[Function]}
  >
    Log In
  </a>
    to post an event!
</h1>
`;
